controllers/v2/fcm_tokens/api.py

```diff
@@ -0,0 +1,89 @@
+import logging
+from repository.fcm_token_repository import FCMTokenRepository
+from flask_restful import Resource, reqparse, marshal_with
+from controllers.v2.v2_blueprint import v2_api
+from services.jwk import requires_auth
+from controllers.v2.fcm_tokens.response_models import response_model
+from repository.user_repository import check_user_exists
+from exception import InvalidTokenError
+from sqlalchemy.exc import SQLAlchemyError
+
+
+parser = reqparse.RequestParser()
+parser.add_argument('token', type=str, required=True, help ="Token must be provided.")
+parser.add_argument('device_type', type=str, required=True, help ="DeviceType must be provided.")
+
+
+class FCMToken(Resource):
+
+    token_repository: FCMTokenRepository
+
+    def __init__(self, token_repository: FCMTokenRepository = FCMTokenRepository()):
+        self.token_repository = token_repository
+
+    @requires_auth
+    @marshal_with(response_model)
+    def post(self, user_id: int):
```

Comment:
You're not actually checking if the user_id in the URL refers to the authenticated user, ditto with delete

Reply:
You should be able to check this with the function `JWKService.decode_user_id()`. Throw a 403 if they don't match.